src/Apps/Order/Components/BankAccountPicker.tsx

```diff
@@ -62,6 +62,31 @@ export const BankAccountPicker: FC<Props> = props => {
 
   const { submitMutation: setPaymentMutation } = useSetPayment()
 
+  useEffect(() => {
+    if (bankAccountSelection) {
+      return
+    }
+
+    if (selectedPaymentMethod === "SEPA_DEBIT") {
+      setBankAccountSelection({ type: "new" })
+    } else if (selectedPaymentMethod === "US_BANK_ACCOUNT") {
+      if (order.bankAccountId) {
+        setBankAccountSelection({
+          type: "existing",
+          id: order.bankAccountId,
+        })
+      } else if (bankAccountsArray.length > 0) {
+        setBankAccountSelection({
+          type: "existing",
+          id: bankAccountsArray[0]?.internalID!,
+        })
+      } else {
+        setBankAccountSelection({ type: "new" })
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I would like to see how I can remove this. In its current form, the hook depends on `order` and `bankAccountsArray`.

Reply:
Generally, you want to just include everything it specifies.

Reply:
Is there an issue with including `order` and `bankAccountsArray` in the dependency array?

Reply:
The `if (bankAccountSelection) { return }` safeguards the user's selection from being overridden on rerender, so it may be safe to add them.